method/stardust/stardust.r

```diff
@@ -0,0 +1,226 @@
+  )
+)
+
+# Description
+description <- "stardust: SNN + louvain on integrated minkowski distance matrices of PCA and spatial coordinates"
+
+opt_parser <- OptionParser(
+  usage = description,
+  option_list = option_list
+)
+opt <- parse_args(opt_parser)
+
+out_dir <- opt$out_dir
+
+# Output files
+label_file <- file.path(out_dir, "domains.tsv")
+embedding_file <- file.path(out_dir, "embedding.tsv")
+# if additional output files are required write it also to out_dir
+
+# Use these filepaths as input ...
+coord_file <- opt$coordinates
+feature_file <- opt$features
+observation_file <- opt$observations
+
+if (!is.na(opt$neighbors)) {
+  neighbors_file <- opt$neighbors
+}
+if (!is.na(opt$matrix)) {
+  matrix_file <- opt$matrix
+}
+if (!is.na(opt$dim_red)) {
+  dimred_file <- opt$dim_red
+}
+if (!is.na(opt$image)) {
+  image_file <- opt$image
+}
+if (!is.na(opt$config)) {
+  config_file <- opt$config
+  config <- fromJSON(config_file)
+}
+
+
+# You can get SpatialExperiment directly
+get_SpatialExperiment <- function(
+    feature_file,
+    observation_file,
+    coord_file,
+    matrix_file = NA,
+    reducedDim_file = NA,
+    assay_name = "counts",
+    reducedDim_name = "reducedDim") {
+  rowData <- read.delim(feature_file, stringsAsFactors = FALSE, row.names = 1)
+  colData <- read.delim(observation_file, stringsAsFactors = FALSE, row.names = 1)
+
+  coordinates <- read.delim(coord_file, sep = "\t", row.names = 1)
+  coordinates <- as.matrix(coordinates[rownames(colData), ])
+  coordinates[,c(1:2)] <- as.numeric(coordinates[,c(1:2)])
+    
+  spe <- SpatialExperiment::SpatialExperiment(
+    rowData = rowData, colData = colData, spatialCoords = coordinates
+  )
+
+  if (!is.na(matrix_file)) {
+    assay(spe, assay_name, withDimnames = FALSE) <- as(Matrix::t(Matrix::readMM(matrix_file)), "CsparseMatrix")
+  }
+
+  # Filter features and samples
+  if ("selected" %in% colnames(rowData(spe))) {
+    spe <- spe[as.logical(rowData(spe)$selected), ]
+  }
+  if ("selected" %in% colnames(colData(spe))) {
+    spe <- spe[, as.logical(colData(spe)$selected)]
+  }
+
+  if (!is.na(reducedDim_file)) {
+    dimRed <- read.delim(reducedDim_file, stringsAsFactors = FALSE, row.names = 1)
+    reducedDim(spe, reducedDim_name) <- as.matrix(dimRed[colnames(spe), ])
+  }
+  return(spe)
+}
+
+# Study configuration
+technology <- opt$technology
+n_clusters <- opt$n_clusters
+method <- config$method
+npcs <- config$npcs
+
+# Seed
+seed <- opt$seed
+set.seed(seed)
+
+## Your code goes here
+
+# SpatialExperiment
+spe <- get_SpatialExperiment(
+    feature_file = feature_file,
+    observation_file = observation_file,
+    coord_file = coord_file,
+    matrix_file = matrix_file#,
+    #reducedDim_file = dimred_file
+)
+
+## Your code goes here
+countMatrix <- as.data.frame(SummarizedExperiment::assay(spe, "counts"))
+spotPositions <- as.data.frame(SummarizedExperiment::colData(spe)[,c("row", "col")])
```

Comment:
If you use row, col this limits technology to ST and VIsium, which should be checked

Reply:
Their paper only describes use in Visium data, but I cannot see why it shouldn't work somewhat with molecule-resolution data, so I changed it to be able to use both.